### PR TITLE
[TextFields] States example

### PR DESCRIPTION
**Pull request by willlarche**
Revealed a couple bugs. Worth it!
simulator screen shot - iphone 7 - 2017-12-13 at 18 22 46

### CODE REVIEW

**Review by samnm — APPROVED** [review]
It appears there are some visual errors when switching styles, but I assume that is unrelated to this change. It doesn't fully clean up from the previous style before changing to the next.

File: components/TextFields/examples/TextFieldStatesViewController.m [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+                        views:@{
+                                @"scrollView" : self.scrollView
+                                }]];
+
+  CGFloat marginOffset = 16;
+  UIEdgeInsets margins = UIEdgeInsetsMake(0, marginOffset, 0, marginOffset);
+  self.scrollView.layoutMargins = margins;
+
+  UITapGestureRecognizer *tapRecognizer =
+  [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(tapDidTouch)];
```
Comment (samnm) [inline components/TextFields/examples/TextFieldStatesViewController.m:225]: indent

File: components/TextFields/examples/TextFieldStatesViewController.m [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+                               attribute:NSLayoutAttributeTrailingMargin
+                              multiplier:1
+                                constant:0]
+  .active = YES;
+
+  NSDictionary *views = @{@"enabled": self.enabled,
+                          @"disabled": self.disabled,
+                          @"errored": self.errored,
+                          @"erroredDisabled": self.erroredDisabled};
+  [NSLayoutConstraint activateConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:[enabled]-[disabled]-[errored]-[erroredDisabled]" options:NSLayoutFormatAlignAllLeading | NSLayoutFormatAlignAllTrailing metrics:nil views:views]];
```
Comment (romoore) [inline components/TextFields/examples/TextFieldStatesViewController.m:106]: Nit: Please break up this line.

File: components/TextFields/examples/TextFieldStatesViewController.m [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+        [(id <MDCTextInputController>)[self.selectedClass alloc] initWithTextInput:self.disabled];
+    self.disabledController.placeholderText = @"Disabled";
+
+    self.erroredController =
+        [(id <MDCTextInputController>)[self.selectedClass alloc] initWithTextInput:self.errored];
+    self.erroredController.placeholderText = @"Errored";
+    [self.erroredController setErrorText:@"Errored" errorAccessibilityValue:nil];
+
+    self.erroredDisabledController =
+        [(id <MDCTextInputController>)[self.selectedClass alloc] initWithTextInput:self.erroredDisabled];
```
Comment (romoore) [inline components/TextFields/examples/TextFieldStatesViewController.m:191]: Nit: Too long as well.

File: components/TextFields/examples/TextFieldStatesViewController.m [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+}
+
+- (void)setupScrollView {
+  self.scrollView = [[UIScrollView alloc] initWithFrame:CGRectZero];
+  self.scrollView.translatesAutoresizingMaskIntoConstraints = NO;
+  [self.view addSubview:self.scrollView];
+  self.scrollView.backgroundColor = UIColor.whiteColor;
+
+  [NSLayoutConstraint
+   activateConstraints:[NSLayoutConstraint
```
Comment (romoore) [inline components/TextFields/examples/TextFieldStatesViewController.m:204]: Nit: Indentation

File: components/TextFields/examples/TextFieldStatesViewController.m [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+                        constraintsWithVisualFormat:@"V:|[scrollView]|"
+                        options:0
+                        metrics:nil
+                        views:@{
+                                @"scrollView" : self.scrollView
+                                }]];
+  [NSLayoutConstraint
+   activateConstraints:[NSLayoutConstraint
```
Comment (romoore) [inline components/TextFields/examples/TextFieldStatesViewController.m:212]: Nit: Indentation

**Review by romoore — APPROVED** [review]
The enabled and disabled states look surprisingly similar.

**Comment by willlarche** [comment]
@samn Yeah that was one of the bugs it surfaced. Not related to the example. 

@romoore These styles were built before final states designs. I’m using this example to check for diffs.

**Pull request merged**